src/http-client.js

```diff
@@ -105,12 +105,13 @@ export class HttpClient {
         if (Response.prototype.isPrototypeOf(result)) {
           response = result;
         } else if (Request.prototype.isPrototypeOf(result)) {
+          request = Promise.resolve(result);
           response = fetch(result);
         } else {
           throw new Error(`An invalid result was returned by the interceptor chain. Expected a Request or Response instance, but got [${result}]`);
         }
 
-        return processResponse(response, this.interceptors);
```

Comment:
But can't result be a Response if request or requestError return a Response directly and we'd then be passing a Response as a Request to response/responseError?